### PR TITLE
Allow sorting self-profile query data by deltas

### LINKED ISSUE #714: Allow sorting by "Time Delta" and "Executions delta"
On a perf diff page (e.g. https://perf.rust-lang.org/detailed-query.html?commit=b48ede19511f5daf372fcf6865f5b798ee775846&base_commit=0820e54a8ad7795d7b555b37994f43cfe62356d4&benchmark=tokio-webpush-simple-check&run_name=full), the "Time delta" and "Executions delta" headers are not clickable like the other headers.

### PR DESCRIPTION
**Pull request by tgnottingham**
Fixes #714.

### CODE REVIEW

**Comment by tgnottingham** [comment]
Enables sorting by query stat deltas (e.g. time) versus another commit. Sorts by actual delta value rather than percent, given that for the deltas we care about, a large delta is generally more important than a large percentage delta (e.g. we don't usually care if a time increases 500% from 0.00001s to 0.00005s).

sort-desc

Swapping commits and reversing sort direction:

sort-asc

File: site/src/api.rs [hunk 1]
```diff
@@ -280,13 +279,26 @@ pub mod self_profile {
     #[derive(Serialize, Deserialize, Clone, Debug)]
     pub struct QueryData {
         pub label: QueryLabel,
-        pub self_time: Duration,
+        pub self_time: u64,
```
Comment (Mark-Simulacrum) [inline site/src/api.rs:282]: Please add a comment like // Seconds. or whatever the unit here is.

**Comment by Mark-Simulacrum** [comment]
Looks good to me modulo one nit.

File: site/src/api.rs [hunk 2]
```diff
@@ -280,13 +279,31 @@ pub mod self_profile {
     #[derive(Serialize, Deserialize, Clone, Debug)]
     pub struct QueryData {
         pub label: QueryLabel,
-        pub self_time: Duration,
+        // Nanoseconds
```
Comment (rylev) [inline site/src/api.rs:282]: FWIW: I generally prefer new types to comments in these cases:
```rust
#[derive(Serialize, Deserialize, Clone, Debug)]
pub struct Nanoseconds(u64);
```

**Pull request merged**